packages/angular/ssr/src/render.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { ApplicationRef, StaticProvider, ɵConsole, ɵresetCompiledComponents } from '@angular/core';
+import {
+  ɵSERVER_CONTEXT as SERVER_CONTEXT,
+  renderApplication,
+  renderModule,
+} from '@angular/platform-server';
+import type { AngularServerApp } from './app';
+import { REQUEST, REQUEST_CONTEXT, RESPONSE_INIT } from './tokens';
+
+/**
+ * Renders an Angular server application to produce a response for the given HTTP request.
+ * Supports server-side rendering (SSR), static site generation (SSG), or app shell rendering.
+ *
+ * @param {AngularServerApp} app - The server application instance to render.
+ * @param {Request} request - The incoming HTTP request object.
+ * @param {'ssr' | 'ssg' | 'app-shell'} serverContext - Context specifying the rendering mode.
+ * @param {unknown} [requestContext] - Optional additional context for the request, such as metadata.
+ * @returns {Promise<Response>} A promise that resolves to a response object representing the rendered content.
+ */
+export async function render(
+  app: AngularServerApp,
+  request: Request,
+  serverContext: 'ssr' | 'ssg' | 'app-shell',
+  requestContext?: unknown,
+): Promise<Response> {
+  const isSsrMode = serverContext === 'ssr';
+  const responseInit: ResponseInit = {};
+  const platformProviders: StaticProvider = [
+    {
+      provide: SERVER_CONTEXT,
+      useValue: serverContext,
+    },
+  ];
+
+  if (isSsrMode) {
+    platformProviders.push(
+      {
+        provide: REQUEST,
+        useValue: request,
+      },
+      {
+        provide: REQUEST_CONTEXT,
+        useValue: requestContext,
+      },
+      {
+        provide: RESPONSE_INIT,
+        useValue: responseInit,
+      },
+    );
+  }
+
+  const { manifest, hooks, isDevMode } = app;
+
+  if (isDevMode) {
+    // Need to clean up GENERATED_COMP_IDS map in `@angular/core`.
+    // Otherwise an incorrect component ID generation collision detected warning will be displayed in development.
+    // See: https://github.com/angular/angular-cli/issues/25924
+    ɵresetCompiledComponents();
+
+    // An Angular Console Provider that does not print a set of predefined logs.
+    platformProviders.push({
+      provide: ɵConsole,
+      useFactory: () => {
+        class Console extends ɵConsole {
+          private readonly ignoredLogs = new Set(['Angular is running in development mode.']);
```

Comment:
> Question: I assume we want to do this because otherwise it prints for every render in dev mode? And it's also not easily feasible to move the offending log statement somewhere it would only apply in the browser?

Yes. 

> Could we just gate it on isPlatformBrowser() or something simple like that? Copying the log statement here feels a bit brittle to changes in the log statement.

It still make sense to have this log on the server when running in production but `ngDevMode` is `true`.